openmc/bounding_box.py

```diff
@@ -87,3 +87,24 @@ def extent(self):
     @property
     def width(self):
         return self.upper_right - self.lower_left
+
+    def extend(self, padding_distance: float) -> 'openmc.BoundingBox':
+        """Returns an extended bounding box
+
+        padding_distance : float
+            The distance to enlarge the bounding box by
+        """
+
+        return BoundingBox(np.array(
+            [
+                self[0][0] - padding_distance,
+                self[0][1] - padding_distance, 
+                self[0][2] - padding_distance
+            ]
+        ), np.array(
+            [
+                self[1][0] + padding_distance,
+                self[1][1] + padding_distance, 
+                self[1][2] + padding_distance
+            ]
+        ))
```

Comment:
```suggestion
        self[0] -= padding_distance
        self[1] += padding_distance
```

I stared at this a bit to figure out why the suggestion above wouldn't work -- because this class inherits from tuple. I'm feeling we should inherit from a mutable type for this object. From working with bounding volume hierarchies, mutable boxes allow for really convenient operations like `bbox.update(other_bbox)` or `bbox.update(point)` to make sure the box contains another box or other points, respectively. With an immutable box, we end up creating potentially more objects than we need b/c we can't modify the underlying data.

If you agree @shimwell, this can definitely be handled in another PR and we can leave this method as-is for now. Just running it by you and @eepeterson to see what you think.

Reply:
I agree with the benefits of mutability here, but don't see any clear need for inheriting from anything in particular in this case. I think we could get away just having a private `_data` attribute that houses the `2x3 np.ndarray` that represents the bounding box.

Reply:
I would be keen on anything that allows us to check if a point is inside the bounding box

Reply:
Not quite what that method would do, but we can add that in! Let's tackle that separately from this capability though. Thank you for this update @shimwell! I think it cleans up the padding process a bit.